### PR TITLE
[T2EMBED] Fix spec file for x64

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Remove stdcall annotated exports from x64

### CODE REVIEW

**Review by HBelusca — APPROVED** [review]
Weird that these functions are also exported in decorated form :thinking:

File: dll/win32/t2embed/t2embed.spec [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
-@ stdcall _TTIsEmbeddingEnabled@8(long ptr) TTIsEmbeddingEnabled
-@ stdcall _TTIsEmbeddingEnabledForFacename@8(str ptr) TTIsEmbeddingEnabledForFacename
-@ stdcall _TTLoadEmbeddedFont@40(ptr long ptr long ptr ptr ptr wstr str ptr) TTLoadEmbeddedFont
-@ stub _TTRunValidationTests@8
+@ stdcall -i386 _TTGetEmbeddingType@8(long ptr) TTGetEmbeddingType
+@ stdcall -i386 _TTIsEmbeddingEnabled@8(long ptr) TTIsEmbeddingEnabled
+@ stdcall -i386 _TTIsEmbeddingEnabledForFacename@8(str ptr) TTIsEmbeddingEnabledForFacename
+@ stdcall -i386 _TTLoadEmbeddedFont@40(ptr long ptr long ptr ptr ptr wstr str ptr) TTLoadEmbeddedFont
+@ stub -i386 _TTRunValidationTests@8
 @ stdcall TTEmbedFontEx(ptr long long ptr ptr ptr ptr ptr long long ptr)
```
Comment (JoachimHenze) [inline dll/win32/t2embed/t2embed.spec:23]: Please allow me to learn something:
"Remove stdcall annotated exports from x64" <- you obviously did not chose to add the -i386 after all that are stdcall. I am pretty sure, that you know what you are doing, so allow me to understand: What is the additional condition that must be met, that it has to be placed here?
-386 will not export them for x64. So is the additional condition, the 'secret' knowledge that I do lack, just that those exports do not exist on x64? Or is there more about it? Thx!

Comment (SergeGautherie) [reply dll/win32/t2embed/t2embed.spec:23]: Looking at it, my guess would be that the excluded ones are forwarders.
If so, another question could be: why exclude `_TTRunValidationTests@8` but not the other decorated stubs?
I am not saying anything is wrong either, only confirming that PR purpose is a bit unclear.

Comment (HBelusca) [reply dll/win32/t2embed/t2embed.spec:23]: The keyword was "annotated" (exports), not "stdcall". The `_Whatever@8(...)` type of decoration is x86 32-bits specific.

Comment (SergeGautherie) [reply dll/win32/t2embed/t2embed.spec:23]: > `_Whatever@8(...)` type of decoration is x86 32-bits specific.

Thanks for the more explicit explanation.
Then, `_TTEnableEmbeddingForFacename@8`, at least, should be excluded too, shouldn't it?

**Review by HBelusca — APPROVED** [review]

File: dll/win32/t2embed/t2embed.spec [hunk 2]
```diff
@@ -9,17 +9,17 @@
...
-@ stdcall _TTEmbedFont@44(long long long ptr ptr ptr ptr ptr long long ptr) TTEmbedFont
-@ stub _TTEmbedFontFromFileA@52
-@ stub _TTEnableEmbeddingForFacename@8
-@ stub _TTGetEmbeddedFontInfo@28
-@ stdcall _TTGetEmbeddingType@8(long ptr) TTGetEmbeddingType
-@ stdcall _TTIsEmbeddingEnabled@8(long ptr) TTIsEmbeddingEnabled
-@ stdcall _TTIsEmbeddingEnabledForFacename@8(str ptr) TTIsEmbeddingEnabledForFacename
-@ stdcall _TTLoadEmbeddedFont@40(ptr long ptr long ptr ptr ptr wstr str ptr) TTLoadEmbeddedFont
-@ stub _TTRunValidationTests@8
+@ stdcall -i386 _TTCharToUnicode@24(ptr ptr long ptr long long) TTCharToUnicode
```
Comment (SergeGautherie) [inline dll/win32/t2embed/t2embed.spec:12]: Note: Forwarder argument lists could be omitted, couldn't they?

Comment (tkreuzer) [reply dll/win32/t2embed/t2embed.spec:12]: In deed it does. Testing this I found a problem though: MSVC removes decorated exports as soon as there is an undecorated export of the same function. `_declspec(dllexport)` doesn't fix this either. Anyway, I'll remove the forwarders.

Comment (SergeGautherie) [reply dll/win32/t2embed/t2embed.spec]: Ah...
Now or later, you may want to check the following cases, for example, for potential issues:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=%40%5B0-9%5D%2B%5C%28&sr=1

**Pull request merged**